src/entity.rs

```diff
@@ -134,17 +134,17 @@ where
     /// Build the [EventSourcedEntity] by loading and applying the current events.
     pub async fn build(self, id: E::Id, pool: Pool) -> Result<EventSourcedEntity<E, L>, Error> {
         let events = current_events_by_id::<E>(&id, &pool).await;
-        let entity = events
-            .try_fold(self.entity, |mut state, (_, event)| {
+        let (version, entity) = events
+            .try_fold((0u64, self.entity), |(_, mut state), (version, event)| {
```

Comment:
Could we start with `None` here and not use `get` below, i.e. work with the `Option<NonZeroU...>`? And use the identifier `last_version`? Then line 147 could be simply rewritten as `last_version`.

Reply:
True, much simpler.